### PR TITLE
Allow empty string as key

### PR DESCRIPTION
**Pull request by ZiyanShi**
The empty string as key acts normally in KVDK. No need to check for zero-sized keys.
Added a file debug.cpp under tests directory and is compiled to dbdebug.
This file is almost a copy of tutorial.cpp but for debugging purpose, because it's very simple and easy to change.

### CODE REVIEW

**Comment by KVKIT** [comment]
The code looks good to me. 
The only question is do we really need to have a debug.cpp and track it in repo?

**Comment by ZiyanShi** [comment]
now only kv_engine.hpp and test.cpp is changed

File: tests/tests.cpp [hunk 1]
```diff
@@ -224,6 +224,27 @@ TEST_F(EngineBasicTest, TestBasicSortedOperations) {
...
+      std::string key{""};
+      std::string value{"Empty key but with value"};
+      std::string sink{};
+      ASSERT_EQ(engine->SSet(overall_skiplist, key, value), Status::Ok);
+      ASSERT_EQ(engine->SGet(overall_skiplist, key, &sink), Status::Ok);
+      ASSERT_EQ(value, sink);
+      ASSERT_EQ(engine->SDelete(overall_skiplist, key), Status::Ok);
+    }
+
+    if (id == 0) {
```
Comment (JiayuZzz) [inline tests/tests.cpp:238]: 1. Every threads should write an empty key to thread_skiplist, the values should be different, then check get result.
2. After SDelete, check if SGet returns not found

File: tests/tests.cpp [hunk 2]
```diff
@@ -326,6 +353,16 @@ TEST_F(EngineBasicTest, TestBasicHashOperations) {
...
       AssignData(v2, v2_len);
 
+      if (id == 0) {
+        std::string key{""};
+        std::string value{"Empty key but with value"};
+        std::string sink{};
+        ASSERT_EQ(engine->Set(key, value), Status::Ok);
+        ASSERT_EQ(engine->Get(key, &sink), Status::Ok);
+        ASSERT_EQ(value, sink);
+        ASSERT_EQ(engine->Delete(key), Status::Ok);
```
Comment (JiayuZzz) [inline tests/tests.cpp:363]: check if Get returns not found after Delete

**Review by JiayuZzz — CHANGES_REQUESTED** [review]

File: tests/tests.cpp (hunk 2 above)
Comment (ZiyanShi) [reply tests/tests.cpp]: add ASSERT_EQ for Get after Delete.

File: tests/tests.cpp (hunk 1 above)
Comment (ZiyanShi) [reply tests/tests.cpp]: change overall_skiplist to global_skiplist, t_skiplist to thread_local_skiplist to avoid confusion.
add test for thread_local_skiplist for empty strings as key.

**Comment by JiayuZzz** [comment]
@ZiyanShi THX, LGTM

BTW, Remember sign-off next time
